js/activity.js

```diff
@@ -452,10 +452,10 @@ class Activity {
             // Append the closeButtonDiv to the helpfulSearchDiv
             this.helpfulSearchDiv.appendChild(closeButtonDiv);
 
-            // Add event listener to remove the search div from the DOM when clicked on the close button
-            closeButton.addEventListener("click", () => {
-                this.helpfulSearchDiv.parentNode.removeChild(this.helpfulSearchDiv); // Remove from DOM
-            });
+            // Add event listener to remove the search div from the DOM
+            const modeButton = docById("begIconText");
+            closeButton.addEventListener("click", this._hideHelpfulSearchWidget);
+            modeButton.addEventListener("click", this._hideHelpfulSearchWidget);
```

Comment:
what does this even accomplish? I assume the solution is due to the missing condition from L503